scripts/repo-tool

```diff
@@ -0,0 +1,159 @@
+################################################################################
+
+. config/options
+
+update_addons_xml() {
+  echo "[*] cleanup addons ..."
+  olddir=""
+  find target/addons/$ADDON_VERSION -iname 'changelog*.txt' | sort -rV | while read line ; do
+    dir=$(dirname $line)
+    if [ "$olddir" = "$dir" ] ; then
+      rm -f $line
+    fi
+    olddir=$dir
+  done
+
+  olddir=""
+  find target/addons/$ADDON_VERSION -iname '*.zip' | sort -rV | while read line ; do
+    dir=$(dirname $line)
+    if [ "$olddir" = "$dir" ] ; then
+      rm -f $line
+    fi
+    olddir=$dir
+  done
+
+  echo "[*] updating addons.xml* ..."
+  rm -rf .addons
+  pwd=`pwd`
+  find target/addons/$ADDON_VERSION -iname addons.xml | while read line ; do
+    localdir=`echo $line | sed s/addons.xml//g`
+    echo " [*] updating $line..."
+    echo '<?xml version="1.0" encoding="UTF-8"?>
+<addons>
+' > $line.tmp
+    for zip in $localdir/*/*.zip ; do
+      mkdir -p ".addons/$localdir"
+      unzip $zip "*/addon.xml" -d ".addons/$localdir" &>/dev/null
+    done
+    find .addons/$localdir -iname addon.xml | grep -v resources/ | while read xml ; do
+      cat $xml | grep -v "<?" >> $line.tmp
+    done
+    echo '
+</addons>' >> $line.tmp
+    mv $line.tmp $line
+    cd $localdir
+
+    md5sum addons.xml > addons.xml.md5
+    cd $pwd
+  done
+  rm -rf .addons
+}
+
+touch_addons_xml() {
+  for PROJECT in Generic RPi2 WeTek_Play; do
+    for archfile in projects/$PROJECT/linux/linux.*.conf ; do
+      ARCH=`echo $archfile | sed -n '$s/\.conf//;$s/.*\.//p'`
+      if [ ! -d target/addons/$ADDON_VERSION/$PROJECT/$ARCH ]; then
+        break
+      fi
+      if [ ! -f target/addons/$ADDON_VERSION/$PROJECT/$ARCH/addons.xml ]; then
+        touch target/addons/$ADDON_VERSION/$PROJECT/$ARCH/addons.xml
+      fi
+    done
+  done
+}
+
+upload() {
+  if [ -f .work/repoconfig ] ; then
+    . .work/repoconfig
+  fi
+  if [ -z "$RSYNC_REPO" ] ; then
+    echo "*** ERROR: \$RSYNC_REPO not set. see .work/repoconfig ***"
+    exit 0
+  fi
+  touch_addons_xml
+  update_addons_xml
+  rsync -av --progress --delete target/addons/$ADDON_VERSION $RSYNC_REPO
+}
+
+build() {
+  for PROJECT in Generic RPi2 WeTek_Play; do
+    for archfile in projects/$PROJECT/linux/linux.*.conf ; do
+      ARCH=`echo $archfile | sed -n '$s/\.conf//;$s/.*\.//p'`
+      for package in $(find $1* -iname package.mk) ; do
+        . $package
+        if [ "$PKG_IS_ADDON" = "yes" ] ; then
+          ADDON=$PKG_NAME
+          PROJECT=$PROJECT ARCH=$ARCH ./scripts/create_addon $ADDON
+        fi
+      done
+    done
+  done
+}
+
+build_projects() {
+  for PROJECT in Generic RPi2 WeTek_Core; do
+    for archfile in projects/$PROJECT/linux/linux.*.conf ; do
+      ARCH=`echo $archfile | sed -n '$s/\.conf//;$s/.*\.//p'`
+      PROJECT=$PROJECT ARCH=$ARCH make
+    done
+  done
+}
+
+update_revision() {
+  for package in $(find $1 -iname package.mk) ; do
+    . $package
```

Comment:
I'd source this in a sub-shell to avoid overwriting any local variables, ie.:
```
+    (
+    . $package
+    if [ "$PKG_IS_ADDON" = "yes" ] ; then
+      sed -i -e "s|PKG_REV=.*|PKG_REV=\"$((PKG_REV+1))\"|" $package
+    fi
+    )
```